### PR TITLE
Implemented client-side APIs to ban/unban packages

### PR DESCRIPTION
**Pull request by Benglin**
### Purpose

This pull request implements client-side API that works with [its REST counterparts](https://github.com/DynamoDS/GReg/pull/2). More details can be found on that pull request.

NOTE: We may or may not require this, it all depends on how we expose the functionality of banning a package. This should only be allowed for internal teammates, so we may end up exposing a web-page that allows banning/unbanning a package.

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@pboyer, please have a look. If you have an idea of how we expose this banning functionality to only internal teammates (e.g. some site served only on ADS domain), then I'd love to hear from you. Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,3 +19,11 @@ GRegClient is a client for the GReg package manager written in C#.  You can find
 
 * magically find all of the dependencies your package has (that's up to you)
 * provide Autodesk 360 login credentials
+
+#### Steps to build GRegClientNET solution:
+
+1. Navigate to `GRegClientNET\third_party\RestSharp` folder
+2. Build `RestSharp.sln` solution (both `Debug` and `Release` configurations)
+3. Build `GRegClientNET\src\GregClient.sln` solution
+
```
Comment (pboyer) [inline README.md]: Thanks for this.

File: src/GregClient/Requests/BanPackage.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    public class BanPackage : Request
+    {
+        private enum Action { Ban, Unban }
+        private readonly Action banAction;
+        private readonly string packageId;
+
+        public BanPackage(string packageId, bool banPackage)
+        {
+            // both endpoints require authentication by the user
+            this.ForceAuthentication = true;
```
Comment (pboyer) [inline src/GregClient/Requests/BanPackage.cs]: By default, `PUT` and `POST` requests require authentication.  So, I don't believe you need this.

File: src/GregClient/Requests/BanPackage.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace Greg.Requests
+{
+    public class BanPackage : Request
+    {
+        private enum Action { Ban, Unban }
+        private readonly Action banAction;
+        private readonly string packageId;
+
+        public BanPackage(string packageId, bool banPackage)
```
Comment (pboyer) [inline src/GregClient/Requests/BanPackage.cs]: Let's not use the `bool` but instead use your nice `BanPackage.Action` enum as the parameter.  `bool`'s are generally hard to understand when used like this.

**Comment by pboyer** [comment]
Only a few changes and LGTM.  However, we should ask ourself, how is this API going to be invoked.  Is there going to be UI in Dynamo.  Are you going to build some sort of utility for using it?

**Comment by Benglin** [comment]
Thanks for reviewing, @pboyer! I'm merging this in now, I believe I have to manually build and update the DLLs in Dynamo repo?

As for the UI, yes we still need to do two things:

1. Returned packages list should not contain banned packages in it.
2. There needs to be UI from which this API can be called (will discuss with @riteshchandawar tomorrow).

**Comment by Benglin** [comment]
A new user story is created to track this work:

- [MAGN-8150](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8150) As Dynamo team members, we should be given the UI to remove a published package.

**Pull request merged**